src/librustc_mir/borrow_check/nll/type_check/mod.rs

```diff
@@ -1289,21 +1289,32 @@ impl<'a, 'gcx, 'tcx> TypeChecker<'a, 'gcx, 'tcx> {
                     );
                 };
             }
-            StatementKind::AscribeUserType(ref place, variance, box ref c_ty) => {
+            StatementKind::AscribeUserType(ascription) => {
```

Comment:
There is still some question in my mind about what user-provided type annotations mean for mutable places (do they apply to all values the place takes on? only at the point where they were given? etc). But if we *do* keep the `Locations::All` interpretation, then we might just remove the `StatementKind` for `AscribeUserType` altogether, right? Instead, we would just iterate over the `UserTypeAnnotation` array.

Reply:
Resolving this as all the code has changed.